src/fu-device-list.c

```diff
@@ -451,6 +491,9 @@ fu_device_list_add (FuDeviceList *self, FuDevice *device)
 		return;
 	}
 
+	/* adopt any required children, which may or may not already exist */
+	fu_device_list_adopt_children (self, device);
```

Comment:
I think it would be better to do this a little earlier (before all plugins are notified about the new device).  This would mean that plugins will be able to know about devices coming in with a parent attached.

Reply:
So move the adoption into FuEngine rather than the DeviceList? I guess that makes sense.

Reply:
Fixed.